## try/except

Когда в программе возникает исключение, она сразу завершает работу.\
Для работы с исключениями используется конструкция **try/except**:

```Python
try:
    2/0
except ZeroDivisionError:
    print("You can't divide by zero")

### You can't divide by zero
```

Чаще всего можно предсказать, какого рода исключения возникнут во время исполнения программы.

Например, если программа на вход ожидает два числа, а на выходе выдает их сумму, а пользователь ввел вместо одного из чисел строку, появится ошибка TypeError, как в примере выше.

Python позволяет работать с исключениями. Их можно перехватывать и выполнять определенные действия в том случае, если возникло исключение.

>**В блоке except можно не указывать конкретное исключение или исключения**.\
В таком случае будут перехватываться все исключения.\
**Это делать не рекомендуется!**

**Конструкция try работает таким образом**:
- сначала выполняются выражения, которые записаны в блоке try.
- если при выполнения блока try не возникло никаких исключений, блок except пропускается, и выполняется дальнейший код.
- если во время выполнения блока try в каком-то месте возникло исключение, оставшаяся часть блока try пропускается.
- если в блоке except указано исключение, которое возникло, выполняется код в блоке except.
- если исключение, которое возникло, не указано в блоке except, выполнение программы прерывается и выдается ошибка.

In [3]:
n = int(input())

phoneBook = {}

key, val = [], []

# Collecting pairs of data .
for i in range(n):
    x = str(input()).split()
    key.append(x[0])
    val.append(x[1])

# I make the dictionary.
phoneBook = dict(zip(key, val))

# Output the desired request.
for i in range(n):
    try:
        name = input()
        if not name in phoneBook :  # 'a' in my_dict
            print('Not found')
        else:
            print(f'{name}={phoneBook[name]}')
    except: 
        break

 3
 unayklejwm 53561825
 ahiff 57272140
 frlrecdfxo 28001354
 aeccxyrbek 21112785


Not found


 ahiff 57272140


Not found


 100000


Not found


100000
unayklejwm 53561825
ahiff 57272140
frlrecdfxo 28001354
aeccxyrbek 21112785
mlgdk 12405837
iyiyycfngr 23841264
subnwvrqdp 77688226
vhypdqfbdy 67846211
lcnbh 86026675
lxbjpmuhis 85475547
enfifbprov 20172436
hnjctgslxf 18125412
slfoglbvah 47530087
mpsrcmrkon 44365224
jnpslqvlkx 88071512
liwyjbsfkk 50646067
kyrkhaikuf 18535535
ubvfx 43551151
krger 80787433{-truncated-}

Not found
ipwpabdefa=14170412
Not found
toykmviqbo=11872347
mooetebtqt=70658483
fbubawkkhq=10613664
Not found
ejqxn=77321147
Not found
Not found
Not found
Not found
xgfpx=58118174
Not found
odacddldrk=73267266
Not found
iakxugylkc=41875647
Not found
djtelththc=20278224
Not found{-truncated-}

In [28]:
n = int(input())

phoneBook = {}

key, val = [], []

for i in range(n):
    string = str(input()).split()
    key.append(string[0])
    val.append(string[1])
    
phoneBook = dict(zip(key, val))

for i in range(n):
    try:
        name = input()
        if name not  in phoneBook:
            print('Not found')
        else:
            print(f'{name}={phoneBook[name]}')
    except:
        break

 3
 sam 99912222
 tom 11122222
 harry 12299933
 sam


sam=99912222


 i


Not found


 harry


harry=12299933
